src/dataflow/src/sink/kafka.rs

```diff
@@ -912,31 +1165,59 @@ where
                 };
             }
 
-            if is_active_worker
-                && matches!(s.send_state, SendState::Init)
-                // Removing this check would cause us to start writing out END consistency rows as
-                // soon as the sink starts up.  There isn't anything inherently wrong with this but
-                // we need to update testdrive to be able to check a reasonable condition that this
-                // is correct.  That is, td currently checks we start with a BEGIN and then END.
-                // We'd need to be able to do a search for a BEGIN to properly test were we to
-                // remove this check.  This would fix #9514.
-                && s.ready_rows.front().is_some()
-            {
-                if s.transactional {
-                    retry_txn_with_error!(|s, producer| { producer.init_transactions().await });
+            // XXX(chae): is there a way to find out if we're teh active worker without consuming
```

Comment:
We should be able to fix #9514 with this change but because we don't identify which worker is active, we don't properly initialize any of them and so we cannot set an initial write frontier to be able to compact timestamps